### PR TITLE
feat(toolkit): add 'cdk context' command

### LINKED ISSUE #311: Toolkit: support refreshing environmental context
The toolkit should support a verb that will refresh the environmental context (such as availability zones, SSM parameter values).

### PR DESCRIPTION
**Pull request by rix0rrr**
Add a command to view and manage cached context values.

Fixes #311.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: docs/src/context.rst [hunk 1]
```diff
@@ -42,3 +42,52 @@ The |cdk| currently supports the following context providers.
...
+   ├───┼────────────────────────────────────────────────────┼────────────────────────────────────────────────────┤
+   │ 1 │ availability-zones:account=123456789012:region=us- │ [ "us-east-1a", "us-east-1b", "us-east-1c",        │
+   │   │ east-1                                             │ "us-east-1d", "us-east-1e", "us-east-1f" ]         │
+   ├───┼────────────────────────────────────────────────────┼────────────────────────────────────────────────────┤
+   │ 2 │ ssm:account=123456789012:parameterName=/aws/       │ "ami-013be31976ca2c322"                            │
+   │   │ service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_ │                                                    │
+   │   │ 64-gp2:region=us-east-1                            │                                                    │
+   └───┴────────────────────────────────────────────────────┴────────────────────────────────────────────────────┘
+
+   Run cdk context --invalidate KEY_OR_NUMBER to invalidate a context key. It will be refreshed on the next CDK synthesis run.
```
[the remark this answers is not in the document]
Comment (rix0rrr) [reply docs/src/context.rst]: I will call it reset.

File: packages/aws-cdk/lib/commands/context.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  print(`Context value ${colors.blue(key)} invalidated. It will be refreshed on the next SDK synthesis run.`);
+}
+
+function keyByNumber(context: any, n: number) {
+  for (const [i, key] of contextKeys(context)) {
+    if (n === i) {
+      return key;
+    }
+  }
+  throw new Error(`No context key with number: ${n}`);
```
[the remark this answers is not in the document]
Comment (rix0rrr) [reply packages/aws-cdk/lib/commands/context.ts]: This is for a semi-interactive usage, where copying the key is a bother and you simply supply the number. Blindly deleting by key # is dangerous, given that the numbering is not unique.

But yeah, just deleting a nonexistant key will be a no-op.

[the remark this answers is not in the document]
Comment (eladb) [reply packages/aws-cdk/lib/commands/context.ts]: You are right of course!

**Comment by rix0rrr** [comment]
Yas

**Pull request merged**